conans/model/conan_file.py

```diff
@@ -1,6 +1,9 @@
 import os
 from contextlib import contextmanager
-
+try:
+    from typing import Union, Dict, Optional, Tuple
```

Comment:
Yes, that is a bit counter-intuitive. But I don't know how type annotations work yet (just an idea about annotations in code in py3)

Reply:
Yeah it is counter intuitive, but mypy treats the hints as source code (essentially) and seems to require it. I believe typings is supported in all python 3 versions and is included in python 3.6+. It may be available for 2 as well

Reply:
I won't oppose adding the comments but from previous experiences, I know that the unused import would be problematic to maintain and I would vote to not include it. Do the comments without the import work for `Jedi`?